test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/schema/SchemaDuplicateChecker.java

```diff
@@ -0,0 +1,340 @@
+            Iterator<String> it = setOfChecksums.iterator();
+            String currentChecksum = null;
+            String nextChecksum = null;
+            while (it.hasNext()) {
+                currentChecksum = nextChecksum;
+                if (currentChecksum == null) {
+                    currentChecksum = it.next();
+                }
+                if (it.hasNext()) {
+                    nextChecksum = it.next();
+                } else {
+                    break;
+                }
+                final String a = currentChecksum;
+                final String b = nextChecksum;
+
+                String contentsA = getOriginalFileContents(schemaFiles, a);
+                String contentsB = getOriginalFileContents(schemaFiles, b);
+                patches.add(DiffUtils.diff(convertFileContentToLines(contentsA),
+                        convertFileContentToLines(contentsB)));
+            }
+            int index = 0;
+            for (String c : setOfChecksums) {
+                //get all files with this checksum
+                sb.append("   PrettyPrint's sha256: ").append(c).append("\n");
+                List<SchemaFile> filesMatch = schemaFiles
+                        .stream()
+                        .filter(se -> se.getSchemaFileName().equals(filename) && se.getPrettyPrintChecksum().equals(c))
+                        .collect(Collectors.toList());
+                filesMatch.forEach(schemaFile -> {
+
+                    sb.append("   + ");
+                    if (schemaFile.isInsideJar()) {
+                        sb.append(format("[%s]", schemaFile.getContainerFile().getAbsolutePath()));
+                        sb.append(schemaFile.getFile().getPath());
+                    } else {
+                        sb.append(schemaFile.getFile().getAbsolutePath());
+                    }
+
+
+                    sb.append("\n");
+                });
+                sb.append("\n");
+
+                if (index + 1 < setOfChecksums.size()) {
+                    List<Delta> deltas = patches.get(index++).getDeltas();
+
+                    deltas.forEach(e -> sb.append(e.toString()).append("\n"));
+                }
+
+                sb.append("\n");
+            }
+            sb.append("\n\n");
+        });
+        return sb.toString();
+
+    }
+
+    private String getOriginalFileContents(List<SchemaFile> schemaFiles, String hash) {
+        return schemaFiles.stream().filter(e -> e.getPrettyPrintChecksum().equals(hash)).map(SchemaFile::getOriginalFileContents).findAny().get();
+    }
+
+    public List<SchemaFile> listSchemaFilesFromJar(File jar) {
+
+        try {
+            final JarFile jarFile = new JarFile(jar);
+            return listSchemaFilesFromJar(jar, jarFile);
+        } catch (IOException e) {
+            throw new SchemaCheckerException(e);
+        }
+    }
+
+    public List<SchemaFile> listSchemaFilesFromJar(final File jar, final JarFile jarFile) {
```

Comment:
Why `public`?

Reply:
Why `jarFile` as a separate argument? Shouldn't both `jar` and `jarFile` refer to the same jar?